exercises/concept/need-for-speed/test/need_for_speed_test.exs

```diff
@@ -0,0 +1,144 @@
+defmodule NeedForSpeedTest do
+  use ExUnit.Case
+
+  import ExUnit.CaptureIO
+
+  describe "print_race" do
+    @task_id nil
+    test "prints a race with no cars" do
+      race = %NeedForSpeed.Race{
+        title: "Need For Speed 2021",
+        total_distance_in_meters: 350,
+        cars: []
+      }
+
+      io = capture_io(fn -> NeedForSpeed.print_race(race) end)
+
+      assert io == """
+             🏁 Need For Speed 2021 🏁
+             Status: Not Started
+             Distance: 350 meters
+
+             Contestants:
+
+             """
+    end
+
+    @task_id nil
+    test "prints a race with a red car" do
+      car = NeedForSpeed.RemoteControlCar.new(:red, "The Fox")
+
+      race = %NeedForSpeed.Race{
+        title: "Need For Speed 2021",
+        total_distance_in_meters: 500,
+        cars: [car]
+      }
+
+      io = capture_io(fn -> NeedForSpeed.print_race(race) end)
+
+      assert io == """
+             🏁 Need For Speed 2021 🏁
+             Status: Not Started
+             Distance: 500 meters
+
+             Contestants:
+
+               1. #{IO.ANSI.red()}The Fox#{IO.ANSI.default_color()}
+               Distance: 0 meters
+               Battery: Battery at 100%
+
+             """
+    end
+
+    @task_id nil
+    test "prints a race with a blue car" do
+      car = NeedForSpeed.RemoteControlCar.new(:blue, "The Hurricane")
+
+      race = %NeedForSpeed.Race{
+        title: "Need For Speed 2021",
+        total_distance_in_meters: 500,
+        cars: [car]
+      }
+
+      io = capture_io(fn -> NeedForSpeed.print_race(race) end)
+
+      assert io == """
+             🏁 Need For Speed 2021 🏁
+             Status: Not Started
+             Distance: 500 meters
+
+             Contestants:
+
+               1. #{IO.ANSI.cyan()}The Hurricane#{IO.ANSI.default_color()}
+               Distance: 0 meters
+               Battery: Battery at 100%
+
+             """
+    end
+
+    @task_id nil
+    test "prints a race with a green car" do
+      car = NeedForSpeed.RemoteControlCar.new(:green, "The Grasshopper")
+
+      race = %NeedForSpeed.Race{
+        title: "Need For Speed 2021",
+        total_distance_in_meters: 777,
+        cars: [car]
+      }
+
+      io = capture_io(fn -> NeedForSpeed.print_race(race) end)
+
+      assert io == """
+             🏁 Need For Speed 2021 🏁
+             Status: Not Started
+             Distance: 777 meters
+
+             Contestants:
+
+               1. #{IO.ANSI.green()}The Grasshopper#{IO.ANSI.default_color()}
+               Distance: 0 meters
+               Battery: Battery at 100%
+
+             """
+    end
+
+    @task_id nil
```

Comment:
Task ids are nil because the test suite won't even run without having solved all of the tasks.